SalesforceCommerceCloud/pwa-kit
# Node warnings with more readable Node versions

# Description

Feedback from sprint demo:
The node version range is now taken from the engines field of the package.json, which is more readable than seeing semver's representation (>=12.0.0 <13.0.0-0||>=14.0.0 <15.0.0-0).

Follow up to this merged PR #384 

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# How to Test-Drive This PR

First of all, run `npm ci` at the root of the monorepo to rebuild all the packages.

Then try testing the generator:
1. Use an incompatible Node version: `nvm use 13`
2. Generate a project with `node packages/pwa-kit-create-app/scripts/create-mobify-app-dev.js --outputDir /tmp/project`
3. You should see a warning like this `... Your app may not work as expected when deployed to Managed Runtime servers which are compatible with Node ^12.0.0 || ^14.0.0`

Also, try running the dev server from within the 'pwa' package:
1. `npm start`
2. Verify the node warning message

And try running `npm ci` at the root folder too with an incompatible node version.

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
